src/juju/index.js

```diff
@@ -131,20 +84,28 @@ async function connectAndLoginWithTimeout(modelURL, options, duration = 5000) {
   fetches it's full status then logs out of the model and closes the connection.
   @param {String} modelUUID The UUID of the model to connect to. Must be on the
     same controller as provided by the controllerURL`.
+  @param {Object} getState A function that'll return the app redux state.
   @returns {Object} The full model status.
 */
-async function fetchModelStatus(modelUUID) {
+async function fetchModelStatus(modelUUID, getState) {
+  const bakery = getBakery(getState());
   const modelURL = controllerURL.replace("/api", `/model/${modelUUID}/api`);
   let status = null;
-  try {
-    const { conn, logout } = await connectAndLoginWithTimeout(
-      modelURL,
-      generateConnectionOptions()
-    );
-    status = await conn.facades.client.fullStatus();
-    logout();
-  } catch (e) {
-    console.error("timeout, unable to log into model:", modelUUID);
+  // Logged in state is checked multiple times as the user may have logged out
+  // between requests.
+  if (isLoggedIn(getState())) {
+    try {
+      const { conn, logout } = await connectAndLoginWithTimeout(
+        modelURL,
+        generateConnectionOptions(false, bakery)
+      );
+      if (isLoggedIn(getState())) {
```

Comment:
Because the call on 98 is awaited it's entirely possible that you've logged out after the call to log in and before we go to request the full status. This enforces that neither of these actions happen if right before, you're logged in.